### PR TITLE
Remove duplicate points using format.py

### PR DESCRIPTION
**Pull request by fabi321**
This is actually a very widespread "problem". There are many entries
that have 2 points after each other that are identical. This happens
especially often with identical first and last points. This should not
alter the appearance in any way.

I'm not planning to do the actual changes in this pr, but rather in a
follow up PR.

### CODE REVIEW

**Comment by fabi321** [comment]
Running the script results in 1757 changed entries right now

File: tools/formatter.py [hunk 1]
```diff
@@ -298,6 +312,8 @@ def print_(*args, **kwargs):
 	entry = fix_no_protocol_urls(entry)
 	print_("Removing extras...")
 	entry = remove_extras(entry)
+	print_("removing duplicate points")
```
Comment (Hans5958) [inline tools/formatter.py:315]:
```suggestion
	print_("Removing duplicate points...")
```

**Review by Hans5958 — CHANGES_REQUESTED** [review]
Also, don't forget if start point = end point! (or if you decided not to...)

**Comment by fabi321** [comment]
I'm accounting for start point == end point by initially setting last to the first entry, and iterating backwards (thus comparing it to the last entry)

**Comment by fabi321** [comment]
Having the off by one error fixed, I'm now up to 1778 changes in atlas.json

**Comment by Hans5958** [comment]
> I'm accounting for start point == end point by initially setting last to the first entry, and iterating backwards (thus comparing it to the last entry)

Ah, makes sense.

File: tools/formatter.py [hunk 2]
```diff
@@ -120,6 +120,20 @@ def remove_extras(entry: dict):
 
 	return entry
 
+def remove_duplicate_points(entry: dict):
+	"""
+	Removes points from paths that occur twice after each other
+	"""
+	path: list = entry['path']
+	previous: list = path[0]
+	for i in range(len(path)-1, -1, -1):
```
Comment (Hans5958) [inline tools/formatter.py:129]:
```suggestion
	for i in range(len(path)-1, 0, -1):
```

Oof, man, don't do -1. You will remove the first point, and get this:

image

**Review by Hans5958 — CHANGES_REQUESTED** [review]

File: tools/formatter.py (hunk 2 above)
Comment (fabi321) [reply tools/formatter.py:129]: After looking into this, I noticed that I did an error while replacing spaces with tabs, which led to this error.
The range to -1 is correct.
I fixed this bug in the latest commit

Comment (Hans5958) [reply tools/formatter.py:129]: Aye, seems fixed now.

**Review by Hans5958 — APPROVED** [review]
LGTM

**Pull request merged**